src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1269,6 +1271,8 @@ internal bool ValidateSignature(byte[] encodedBytes, byte[] signature, SecurityK
                 return false;
             }
 
+            Validators.ValidateAlgorithm(algorithm, key, securityToken, validationParameters);
```

Comment:
Would it make sense to move this call to the private ValidateSignature method?
Am not sure what we are gaining by putting the check in this method.

Reply:
We currently expect the SecurityKey. The thought being that it could be useful for making a validation decision in the delegate.

Reply:
What would be a potential use case where a SecurityKey is expected/useful? The alg/enc is already defined in a token and TVP holds a collection of valid algorithms.

Reply:
@brentschmaltz can correct me if I am wrong here.

But the biggest one that comes to mind is AdditionalData on the JsonWebKey typed arguments

Reply:
@GeoK @keegan-caruso passing the SecurityKey we are about to use can be helpful as the 'SecurityKey' could be user specified as we have a delegate that the user could have coded into the call graph.

Reply:
I'm not pushing back on the change, just trying to identity a use case.
If the idea is to be future-proof, and provide more extensibility, I'm ok with the change.